lib/krane/kubectl.rb

```diff
@@ -100,6 +100,14 @@ def server_dry_run_enabled?
       server_version >= Gem::Version.new(SERVER_DRY_RUN_MIN_VERSION)
     end
 
+    def dry_run_flag
+      if server_version >= Gem::Version.new("1.18")
+        "--dry-run=server"
+      else
+        "--server-dry-run"
+      end
```

Comment:
We actually need to handle this with the given kubectl client version as well, which might bite us